dfastmi/gui/dialog_view_model.py

```diff
@@ -0,0 +1,296 @@
+        progloc = FileUtils.get_progloc()
+        filename = progloc + os.path.sep + "dfastmi_usermanual.pdf"
+        return filename
+    
+    @property
+    def report(self) -> str:
+        return ApplicationSettingsHelper.get_filename("report.out")
+
+
+    def gui_text(self, key: str, prefix: str = "gui_", dict: Dict[str, Any] = {}):
+        """
+        Query the global dictionary of texts for a single string in the GUI.
+
+        This routine concatenates the prefix and the key to query the global
+        dictionary of texts. It selects the first line of the text obtained and
+        expands and placeholders in the string using the optional dictionary
+        provided.
+
+        Arguments
+        ---------
+        key : str
+            The key string used to query the dictionary (extended with prefix).
+        prefix : str
+            The prefix used in combination with the key (default "gui_").
+        dict : Dict[str, Any]
+            A dictionary used for placeholder expansions (default empty).
+
+        Returns
+        -------
+            The first line of the text in the dictionary expanded with the keys.
+        """
+        cstr = ApplicationSettingsHelper.get_text(prefix + key)
+        application_setting = cstr[0].format(**dict)
+        return application_setting
+    
+    def updated_branch(self, branch_name: str) -> None:
+        """
+        Adjust the GUI for updated branch selection.
+
+        Arguments
+        ---------
+        branch_name : str
+            Newly selected branch.
+        """
+        self.current_branch = self.model.rivers.get_branch(branch_name)
+        self.current_reach = self._current_branch.reaches[0]
+        self.update_qvalues()
+
+
+    def updated_reach(self, reach_name: str) -> None:
+        """
+        Adjust the GUI for updated reach selection.
+
+        Arguments
+        ---------
+        reach_name : str
+            Newly selected reach.
+        """
+        if reach_name:
+            self.current_reach = self._current_branch.get_reach(reach_name)
+            self.update_qvalues()
+
+
+    def update_qvalues(self) -> None:
+        """
+        Adjust the GUI for updated characteristic discharges.
+
+        Arguments
+        ---------
+        None
+        """
+                
+        try:        
+            q_threshold = float(self._current_reach.qstagnant)
+            time_fractions_of_the_year = ConfigurationInitializer.get_time_fractions_of_the_year(self._current_reach.hydro_t)
+            time_mi = ConfigurationInitializer.calculate_time_mi(q_threshold, self._current_reach.hydro_q, time_fractions_of_the_year)
+            celerity = ConfigurationInitializer.get_bed_celerity(self._current_reach, self._current_reach.hydro_q)
+            slength = dfastmi.kernel.core.estimate_sedimentation_length(time_mi, celerity)
+            self.slength = "{:.0f}".format(slength)
+        except:
+            self.slength = "---"
```

Comment:
is this the valid except handling action?

Reply:
yes